### PR TITLE
console: migrate CollapsibleToggle to typescript

### LINKED ISSUE #4314: console: migrate to TypeScript
As we merged https://github.com/hasura/graphql-engine/pull/3902 recently we plan to gradually migrate frontend code to TypeScript.

Console codebase is pretty large, so we would love your help! 💪

### How to contribute?

1. Convert file from `.js` to `.ts` or `.tsx`. 
2. Add types, fix all type errors. 
3. Submit a PR! 🎉

[Here you can find a guide on how to contribute to Hasura console](https://github.com/hasura/graphql-engine/blob/master/console/README.md#contributing-to-hasura-console).

We recommend you to take a look at the codebase and find something that is still in JS and seems interesting for you to migrate. If you find something, just let us know in the comment and go ahead! 😎 

<details>
  <summary>Click to see the list of easy and small modules that are a good fit for the start</summary>

- [ ] [AlertBox component](https://github.com/hasura/graphql-engine/tree/master/console/src/components/UIKit/atoms/AlertBox) @harishm72 https://github.com/hasura/graphql-engine/pull/4721

- [ ] [Tabs component](https://github.com/hasura/graphql-engine/tree/master/console/src/components/UIKit/atoms/Tabs) @sensiblefolk https://github.com/hasura/graphql-engine/pull/4536
- [ ] [Tooltip component](https://github.com/hasura/graphql-engine/tree/master/console/src/components/UIKit/atoms/Tooltip) @ngprnk https://github.com/hasura/graphql-engine/pull/4524

- [ ] [AceEditor](https://github.com/hasura/graphql-engine/tree/master/console/src/components/Common/AceEditor) @sensiblefolk https://github.com/hasura/graphql-engine/pull/4536

- [ ] [Headers component and utils](https://github.com/hasura/graphql-engine/blob/master/console/src/components/Common/Headers) @harishm72 
- [ ] [Routes utils](https://github.com/hasura/graphql-engine/blob/master/console/src/components/Common/utils/routesUtils.j) @ngprnk

- [ ] [Error boundary component](https://github.com/hasura/graphql-engine/blob/master/console/src/components/Error/ErrorBoundary.js) @marekdano 
https://github.com/hasura/graphql-engine/pull/4542
- [ ] ~[SDL utils](https://github.com/hasura/graphql-engine/blob/master/console/src/shared/utils/sdlUtils.js)~ @mubaidr (can't be merged until cli-ext supports TS)
- [ ] ~[Wrapping type utils](https://github.com/hasura/graphql-engine/blob/master/console/src/shared/utils/wrappingTypeUtils.js)~ @Windrushfarer (can't be merged until cli-ext supports TS)
- [ ] ~[GraphQL Schema utils](https://github.com/hasura/graphql-engine/blob/master/console/src/shared/utils/graphqlSchemaUtils.js)~ @Windrushfarer (can't be merged until cli-ext supports TS)
- [ ] ~[Hasura custom types utils](https://github.com/hasura/graphql-engine/blob/master/console/src/shared/utils/hasuraCustomTypeUtils.js)~ @mubaidr (can't be merged until cli-ext supports TS)

- [ ] [Telemetry (Redux actions)](https://github.com/hasura/graphql-engine/tree/master/console/src/telemetry) @marekdano https://github.com/hasura/graphql-engine/pull/4737

- [ ] [Server.js](https://github.com/hasura/graphql-engine/blob/master/console/src/server.js) @Windrushfarer
- [ ] [Routes.js](https://github.com/hasura/graphql-engine/blob/master/console/src/routes.js) @marekdano
- [ ] [Globals.js](https://github.com/hasura/graphql-engine/blob/master/console/src/Globals.js) @marekdano https://github.com/hasura/graphql-engine/pull/4587
- [ ] [QueryBuilderJson](https://github.com/hasura/graphql-engine/blob/master/console/src/components/Common/QueryBuilderJson/QueryBuilderJson.js) @marekdano https://github.com/hasura/graphql-engine/pull/4803
- [ ] [Login Actions](https://github.com/hasura/graphql-engine/blob/master/console/src/components/Login/Actions.js)
- [ ] [Login](https://github.com/hasura/graphql-engine/blob/master/console/src/components/Login/Login.js) @ryands17 https://github.com/hasura/graphql-engine/pull/4794
- [ ] [Main actions](https://github.com/hasura/graphql-engine/blob/master/console/src/components/Main/Actions.js) @marekdano https://github.com/hasura/graphql-engine/pull/4876
- [ ] [Main state](https://github.com/fkhadra/graphql-engine/blob/master/console/src/components/Main/State.js) @marekdano https://github.com/hasura/graphql-engine/pull/4804
- [ ] [About service](https://github.com/hasura/graphql-engine/blob/master/console/src/components/Services/About/About.js) @ryands17 https://github.com/hasura/graphql-engine/pull/4794
- [ ] [Settings Actions](https://github.com/hasura/graphql-engine/blob/master/console/src/components/Services/Settings/Actions.js) @gokulchandra 
- [ ] [Sidebar](https://github.com/hasura/graphql-engine/blob/master/console/src/components/Services/Settings/Sidebar.js) @marekdano https://github.com/hasura/graphql-engine/pull/4690
- [ ] [Settings utils](https://github.com/hasura/graphql-engine/blob/master/console/src/components/Services/Settings/utils.js) @poul-kg https://github.com/hasura/graphql-engine/pull/4684
- [ ] CollapsibleToggle component @kshitijk83 https://github.com/hasura/graphql-engine/pull/4697
</details>

<details>
  <summary>Click to see the list of modules already migrated by the community 🙌</summary>

- [x] [Typography component](https://github.com/hasura/graphql-engine/tree/master/console/src/components/UIKit/atoms/Typography) @kawamataryo https://github.com/hasura/graphql-engine/pull/4373

- [x] [SwitchButton component](https://github.com/hasura/graphql-engine/tree/master/console/src/components/UIKit/atoms/SwitchButton) @kawamataryo https://github.com/hasura/graphql-engine/pull/4504

- [x] [Spinner component](https://github.com/hasura/graphql-engine/tree/master/console/src/components/UIKit/atoms/Spinner) @kawamataryo https://github.com/hasura/graphql-engine/pull/4504

- [x] [VoyagerView](https://github.com/hasura/graphql-engine/blob/master/console/src/components/Services/VoyagerView/VoyagerView.js) @marekdano https://github.com/hasura/graphql-engine/pull/4734

- [x]  [Button component](https://github.com/hasura/graphql-engine/tree/master/console/src/components/UIKit/atoms/Button) @kawamataryo https://github.com/hasura/graphql-engine/pull/4570

- [x] [Settings container](https://github.com/hasura/graphql-engine/blob/master/console/src/components/Services/Settings/Container.js) @ryands17 https://github.com/hasura/graphql-engine/pull/4724
- [x] Cypress setup and tests @rajanand02 https://github.com/hasura/graphql-engine/pull/4550
- [x] [Theme](https://github.com/hasura/graphql-engine/blob/master/console/src/components/UIKit/theme/index.js) @alioguzhan https://github.com/hasura/graphql-engine/pull/4325

- [x]  [Checkbox component](https://github.com/hasura/graphql-engine/tree/master/console/src/components/UIKit/atoms/Checkbox) @kawamataryo https://github.com/hasura/graphql-engine/pull/4373

- [x] [Icon component](https://github.com/hasura/graphql-engine/tree/master/console/src/components/UIKit/atoms/Icon) @kawamataryo https://github.com/hasura/graphql-engine/pull/4373
- [x] [RadioButton component](https://github.com/hasura/graphql-engine/tree/master/console/src/components/UIKit/atoms/RadioButton) @kawamataryo https://github.com/hasura/graphql-engine/pull/4373
- [x] [GraphQL utils](https://github.com/hasura/graphql-engine/blob/master/console/src/components/Common/utils/graphqlUtils.js) @mubaidr https://github.com/hasura/graphql-engine/pull/4519
- [x] [SQL utils](https://github.com/beerose/graphql-engine/blob/master/console/src/components/Common/utils/sqlUtils.js) @mubaidr https://github.com/hasura/graphql-engine/pull/4519
- [x] [Main utils](https://github.com/hasura/graphql-engine/blob/master/console/src/components/Main/utils.js) @mubaidr https://github.com/hasura/graphql-engine/pull/4519

- [x] [Modal component](https://github.com/hasura/graphql-engine/blob/master/console/src/components/Common/Modal/Modal.js) @irshadjsr21 https://github.com/hasura/graphql-engine/pull/4532

- [x] [GqlCompatibilityWarning component](https://github.com/hasura/graphql-engine/blob/master/console/src/components/Common/GqlCompatibilityWarning/GqlCompatibilityWarning.js) @fkhadra https://github.com/hasura/graphql-engine/pull/4544
- [x] [Telemetry filter](https://github.com/hasura/graphql-engine/blob/master/console/src/telemetryFilter.js) @marekdano https://github.com/hasura/graphql-engine/pull/4579
- [x] Common/WarningSymbol and Common/TextInput @fkhadra https://github.com/hasura/graphql-engine/pull/4562
- [x] HTML helpers @ryands17 https://github.com/hasura/graphql-engine/pull/4641
- [x] Route hooks helpers @ryands17 https://github.com/hasura/graphql-engine/pull/4641
- [x] Version utils @ryands17 https://github.com/hasura/graphql-engine/pull/4641
</details>

### Reach out to us!

Feel free to reach if you have questions or need help getting started. You can leave comments here or you can tag me in your PR if you need any help or you're not sure about something!

You can also get in touch on our [Discord](https://t.co/aVWe9KFrqX?amp=1) #contrib channel. 

---

To avoid duplicate work please comment on which part you want to work on (as long as nobody else is working on it) so we can mark it as taken.

### PR DESCRIPTION
**Pull request by kshitijk83**
### Description

This pull request convert [CollapsibleToggle](https://github.com/hasura/graphql-engine/blob/master/console/src/components/Common/CollapsibleToggle/CollapsibleToggle.js) component to typescript
### Changelog

- [ ] `CHANGELOG.md` is updated with user-facing content relevant to this PR.

### Affected components

- [x] Console

### Related Issues

 #4314 

### Solution and Design

added state and props interfaces:
```
interface CollapsibleToggleProps {
  title: string;
  isOpen: boolean;
  toggleHandler?: () => void;
  testId: string;
  children: React.ReactNode;
  useDefaultTitleStyle?: boolean;
}

interface CollapsibleToggleState {
  isOpen: boolean;
  toggleHandler: () => void;
}
```

#### Catalog upgrade

Does this PR change Hasura Catalog version?
- [x] No
- [ ] Yes
  - [ ] Updated docs with SQL for downgrading the catalog 

#### Metadata

Does this PR add a new Metadata feature?
- [x] No
- [ ] Yes
  - Does `run_sql` auto manages the new metadata through schema diffing?
    - [ ] Yes
    - [ ] Not required
  - Does `run_sql` auto manages the definitions of metadata on renaming?
    - [ ] Yes
    - [ ] Not required
  - Does `export_metadata`/`replace_metadata` supports the new metadata added?
    - [ ] Yes
    - [ ] Not required

#### GraphQL
- [x] No new GraphQL schema is generated
- [ ] New GraphQL schema is being generated:
   - [ ] New types and typenames are correlated
   
   

#### Breaking changes

- [x] No Breaking changes
- [ ] There are breaking changes:

### CODE REVIEW

File: console/src/components/Common/CollapsibleToggle/CollapsibleToggle.tsx [hunk 1]
```diff
@@ -9,8 +9,25 @@ import React from 'react';
...
  */
 
-class CollapsibleToggle extends React.Component {
-  constructor(props) {
+interface CollapsibleToggleProps {
+  title: string;
+  isOpen: boolean;
+  toggleHandler?: () => void;
+  testId: string;
+  children: React.ReactNode;
```
Comment (beerose) [inline console/src/components/Common/CollapsibleToggle/CollapsibleToggle.tsx:17]: Typings for `children` will already be there as we are extending `React.Component`.
```suggestion
```

Comment (kshitijk83) [reply console/src/components/Common/CollapsibleToggle/CollapsibleToggle.tsx]: sorry i didn't load vs-code from console folder so eslint rules weren't loaded, i corrected and solved those issues now.

**Comment by kshitijk83** [comment]
@beerose are there any required changes left?

**Comment by beerose** [comment]
> @beerose are there any required changes left?

No, I just approved 🙂
Thanks again for your PR!

**Pull request merged**